### ISSUE TITLE
Finalize `cibuildwheel` tooling

### ISSUE THREAD

**Issue by vytas7** [falconry/falcon#2310]
The cibuildwheel gate is almost ready to rock 🤘.

However, minor cleanup is still needed before we can get 4.0.0a1 out the door in one click:

- [x] ~~Disable or remove the old wheels workflow~~ :arrow_right: deferred to #2311.
- [x] Publish at least `sdist` to the release page (do we really need to also upload all binaries like now?).
- [x] Make sure `sdist` is uploaded first in a separate step, otherwise `PyPI` and `pip` get confused by a release without `sdist` in the interim period.
- [x] Clean up and adapt the script(s) to check the built wheels.
